### PR TITLE
Fetch shows based on user location

### PR DESCRIPTION
**Pull request by l2succes**
Related: 
- https://artsyproduct.atlassian.net/browse/LD-146
- https://artsyproduct.atlassian.net/browse/LD-144

This also ports over the relay mocking tools introduced in reaction with this PR https://github.com/artsy/reaction/pull/1854 over to emission

RFC for react-native-svg: https://github.com/artsy/emission/issues/1273
#skip_new_tests

### CODE REVIEW

**Comment by l2succes** [comment]
Here's the RFC for the new dependency https://github.com/artsy/emission/issues/1273.
I should note we use a PNG in the screenshot above and you can notice the pixelation as you zoom in.

File: Pod/Classes/GraphQL/ARGraphQLQueryCache.m [hunk 1]
```diff
@@ -158,7 +158,9 @@ - (void)clearAll;
     if (response == nil) {
         // this is a sentinel to indicate the value is being fetched and a place where interested parties can request a
         // future resolved value as a promise
-        NSAssert(self.inFlightRequests[cacheKey] == nil, @"[ARGraphQLQueryCache] [%@] Expected no promise queue to exist yet", cacheKey);
+        if (self.inFlightRequests[cacheKey] == nil) {
+            DLog(@"[ARGraphQLQueryCache] [%@] Expected no promise queue to exist yet", cacheKey);
```
Comment (l2succes) [inline Pod/Classes/GraphQL/ARGraphQLQueryCache.m]: this `NSAssert` actually shows up as a red screen in React Native which is the error we were seeing regularly.

Comment (ashfurrow) [reply Pod/Classes/GraphQL/ARGraphQLQueryCache.m]: I ran into this assertion failure while trying to do prefetching for bid flow. I didn't have the chance to go back and dig into it. @alloy does this make change make sense?

File: src/lib/tests/MockRelayRenderer.tsx [hunk 2]
```diff
@@ -2,15 +2,34 @@ import { IMocks } from "graphql-tools/dist/Interfaces"
 import React from "react"
 import { QueryRenderer, RelayContainer } from "react-relay"
 import { Environment, GraphQLTaggedNode, OperationBase, OperationDefaults, RecordSource, Store } from "relay-runtime"
-import { ContextProvider } from "../utils/Context"
+import { ContextConsumer, ContextProvider } from "../utils/Context"
 import renderWithLoadProgress from "../utils/renderWithLoadProgress"
-import { createMockNetworkLayer } from "./createMockNetworkLayer"
+import { createMockNetworkLayer, createMockNetworkLayer2 } from "./createMockNetworkLayer"
```
Comment (ashfurrow) [inline src/lib/tests/MockRelayRenderer.tsx]: `createMockNetworkLayer2`?

File: src/lib/tests/createMockNetworkLayer/__tests__/createMockNetworkLayer-tests.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import { IResolvers } from "graphql-tools/dist/Interfaces"
+import { graphql } from "react-relay"
+import { Environment, fetchQuery, GraphQLTaggedNode, RecordSource, Store } from "relay-runtime"
+import { createMockNetworkLayer2 } from "../index"
+
+describe("createMockNetworkLayer", () => {
```
Comment (ashfurrow) [inline src/lib/tests/createMockNetworkLayer/__tests__/createMockNetworkLayer-tests.tsx]: Could we make sure to add a comment to any files that were copied over from Reaction? It'll make it easier once we publish a package with the mock renderer stuff that both Reaction and Emission can move to 👍

File: Pod/Classes/GraphQL/ARGraphQLQueryCache.m (hunk 1 above)
Comment (alloy) [reply Pod/Classes/GraphQL/ARGraphQLQueryCache.m]: This just works around the assertion being hit, there's definitely something off with the overall codeflow if this is being hit. I'm fine with falling back to the same code that's used in production to ease development, for now, but a ticket should be filed to look into this imo.

Comment (ashfurrow) [reply Pod/Classes/GraphQL/ARGraphQLQueryCache.m]: Gotcha, I've opened a ticket here to track: https://artsyproduct.atlassian.net/browse/GROW-1098

Comment (alloy) [reply Pod/Classes/GraphQL/ARGraphQLQueryCache.m]: Thanks! 👍

File: src/lib/tests/MockRelayRenderer.tsx (hunk 2 above)
Comment (l2succes) [reply src/lib/tests/MockRelayRenderer.tsx]: I just followed the naming from the reaction code but yeah I agree it's a lil cryptic

File: src/lib/tests/createMockNetworkLayer/__tests__/createMockNetworkLayer-tests.tsx (hunk 3 above)
Comment (l2succes) [reply src/lib/tests/createMockNetworkLayer/__tests__/createMockNetworkLayer-tests.tsx]: Yup, definitely will add a link to those

**Comment by l2succes** [comment]
Had to disable two tests to unblock this PR, but left notes to follow back up on getting those running again.

**Pull request merged**